source/gravity_model/function.cc

```diff
@@ -39,8 +39,10 @@ namespace aspect
     gravity_vector (const Point<dim> &position) const
     {
       Tensor<1,dim> gravity;
+      const Utilities::NaturalCoordinate<dim> point =
+        this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system);
       for (unsigned int d=0; d<dim; ++d)
-        gravity[d] = function.value(position,d);
+        gravity[d] = function.value(Utilities::convert_array_to_point<dim>(point.get_coordinates()),d);
```

Comment:
Your code is not wrong, but this one will be slightly faster, because some computations will be done only once instead of `dim` times:

```
const Point<dim> point_in_coordinate_system = Utilities::convert_array_to_point<dim>(point.get_coordinates());
for (unsigned int d=0; d<dim; ++d)
  gravity[d] = function.value(point_in_coordinates,d);
```